homeassistant/components/bsblan/climate.py

```diff
@@ -0,0 +1,239 @@
+        return self._info.device_identification
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement which this thermostat uses."""
+        if self._temperature_unit == "&deg;C":
+            return TEMP_CELSIUS
+        return TEMP_FAHRENHEIT
+
+    @property
+    def supported_features(self) -> int:
+        """Flag supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def hvac_mode(self):
+        """Return the current operation mode."""
+        return self._current_hvac_mode
+
+    @property
+    def hvac_modes(self):
+        """Return the list of available operation modes."""
+        return HVAC_MODES
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    @property
+    def preset_modes(self):
+        """List of available preset modes."""
+        return PRESET_MODES
+
+    @property
+    def preset_mode(self):
+        """Return the preset_mode."""
+        return self._preset_mode
+
+    async def async_set_preset_mode(self, preset_mode):
+        """Set preset mode."""
+        _LOGGER.debug("Setting preset mode to: %s", preset_mode)
+        if preset_mode == PRESET_NONE:
+            # restore previous hvac mode
+            self._current_hvac_mode = self._store_hvac_mode
+        else:
+            # Store hvac mode.
+            self._store_hvac_mode = self._current_hvac_mode
+            await self.async_set_data(preset_mode=preset_mode)
+
+    async def async_set_hvac_mode(self, hvac_mode):
+        """Set HVAC mode."""
+        _LOGGER.debug("Setting HVAC mode to: %s", hvac_mode)
+        # preset should be none when hvac mode is set
+        self._preset_mode = PRESET_NONE
+        await self.async_set_data(hvac_mode=hvac_mode)
+
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperatures."""
+        await self.async_set_data(**kwargs)
+
+    async def async_set_data(self, **kwargs: Any) -> None:
+        """Set device settings using BSBLan."""
+        data = {}
+
+        if ATTR_TEMPERATURE in kwargs:
+            data[ATTR_TARGET_TEMPERATURE] = kwargs[ATTR_TEMPERATURE]
+            _LOGGER.debug("Set temperature data = %s", data)
+
+        if ATTR_HVAC_MODE in kwargs:
+            data[ATTR_HVAC_MODE] = HA_STATE_TO_BSBLAN[kwargs[ATTR_HVAC_MODE]]
+            _LOGGER.debug("Set hvac mode data = %s", data)
+
+        if ATTR_PRESET_MODE in kwargs:
+            # for now we set the preset as hvac_mode as the api expect this
+            data[ATTR_HVAC_MODE] = HA_PRESET_TO_BSBLAN[kwargs[ATTR_PRESET_MODE]]
+
+        try:
+            await self.bsblan.thermostat(**data)
+        except BSBLanError:
+            _LOGGER.error("An error occurred while updating the BSBLan device.")
```

Comment:
We don't end logging messages with period.
```suggestion
            _LOGGER.error("An error occurred while updating the BSBLan device")
```